common/path.cpp

```diff
@@ -178,8 +231,107 @@ Path Path::join(const String &str, char separator) const {
 
 Path Path::join(const char *str, char separator) const {
 	Path temp(*this);
-	temp.joinInPlace(str, DIR_SEPARATOR);
+	temp.joinInPlace(str, separator);
 	return temp;
 }
 
+Common::Array<Common::String> Path::splitComponents() const {
+	Common::Array<Common::String> res;
+	Common::String cur;
+	for (uint i = 0; i < _str.size(); i++) {
+		if (_str[i] == ESCAPER) {
+			i++;
+			if (_str[i] == ESCAPE_SLASH)
+				cur += '/';
+			else if (_str[i] == ESCAPE_SEPARATOR) {
+				res.push_back(cur);
+				cur = "";
+			}
+		} else
+			cur += _str[i];
+	}
+
+	res.push_back(cur);
+
+	return res;
+}
+
+Path Path::punycodeDecode() const {
+	Common::Array<Common::String> c = splitComponents();
+	String res;
+
+	for (uint i = 0; i < c.size(); i++) {
+		res += escapePath(punycode_decodefilename(c[i]));
+		if (i + 1 < c.size())
+			res += DIR_SEPARATOR;
+	}
+
+	Path ret;
+	ret._str = res;
+	return ret;
+}
+
+static String getIdentifierComponent(const String& in) {
```

Comment:
Could you please add a brief comment explaining what "identifier" mean here? Or is it simply "getPathComponent"?

Reply:
Done